pydatastructs/trees/tests/test_binary_trees.py

```diff
@@ -436,3 +436,109 @@ def test_SplayTree():
+        " (None, 90, 0, None, None), (8, 100, 0, None, 10), (None, 110, 0, None, None), (9, 120, 1, None, 13),"
+        " (None, 130, 0, None, None), (12, 140, 0, None, 15), (None, 150, 0, None, None), (14, 160, 1, None, 16),"
+        " (None, 170, 0, None, 17), (None, 180, 1, None, None)]")
+
+    tree.delete(180)
+    tree.delete(130)
+    tree.delete(110)
+    tree.delete(190)
+    assert str(tree) == \
+        ("[(None, 10, 0, None, None), (0, 20, 0, None, 2), (None, 30, 0, None, None), (1, 40, 1, None, 5), (None, 50, 0, None, None),"
+         " (4, 60, 0, None, 6), (None, 70, 0, None, None), (3, 80, 0, None, 11), (None, 90, 1, None, None), (8, 100, 0, None, None), '',"
+         " (9, 120, 0, None, 15), '', (None, 140, 0, None, 14), (None, 150, 1, None, None), (13, 160, 1, None, 16), (None, 170, 0, None, None)]")
+
+    tree.delete(170)
+    tree.delete(100)
+    tree.delete(60)
+    assert str(tree) == \
+        ("[(None, 10, 0, None, None), (0, 20, 1, None, 2), (None, 30, 0, None, None), (1, 40, 0, None, 5), (None, 50, 1, None, None), (4, 70, 0, None, None),"
+         " '', (3, 80, 0, None, 11), (None, 90, 0, None, None), '', '', (8, 120, 0, None, 14), '', (None, 140, 0, None, None), (13, 150, 1, None, 15), (None, 160, 0, None, None)]")
+
+    tree.delete(70)
+    tree.delete(140)
+
+    assert str(tree) == ("[(None, 10, 0, None, None), (0, 20, 1, None, 2), (None, 30, 0, None, None), (1, 40, 0, None, 4), (None, 50, 0, None, None), '', '', (3, 80, 0, None, 11),"
+                         " (None, 90, 0, None, None), '', '', (8, 120, 0, None, 14), '', '', (None, 150, 0, None, 15), (None, 160, 1, None, None)]")
+
+    tree.delete(150)
+    tree.delete(120)
+    assert str(tree) == ("[(None, 10, 0, None, None), (0, 20, 0, None, 2), (None, 30, 0, None, None), (1, 40, 0, None, 7), (None, 50, 0, None, None), '', '', (4, 80, 0, None, 11),"
+                         " (None, 90, 1, None, None), '', '', (8, 160, 0, None, None), '', '', '']")
+
+    tree.delete(50)
+    tree.delete(80)
+    assert str(tree) == ("[(None, 10, 0, None, None), (0, 20, 1, None, 2), (None, 30, 0, None, None), (1, 40, 0, None, 5), '', (None, 90, 0, None, 6), (None, 160, 1, None, None)]")
+
+    tree.delete(30)
+    tree.delete(20)
+    assert str(tree) == ("[(None, 10, 0, None, None), '', '', (0, 40, 0, None, 5), '', (None, 90, 0, None, 6), (None, 160, 1, None, None)]")
+
+    tree.delete(10)
+    assert str(tree) == ("[(None, 40, 0, None, None), (0, 90, 0, None, 2), (None, 160, 0, None, None)]")
+
+    tree.delete(40)
+    tree.delete(90)
+    assert str(tree) == ("[(None, 160, 0, None, None)]")
+
+    tree.delete(160)
+    assert str(tree) == ("[(None, None, 0, None, None)]")
+
+    tree = Redblacktree()
+    tree.insert(50)
+    tree.insert(40)
+    tree.insert(30)
+    tree.insert(20)
+    tree.insert(10)
+    tree.insert(5)
+
+    tree.delete(50)
+    tree.delete(20)
+    tree.delete(30)
```

Comment:
Please provide a reference for these tests so that it's easy to verify their correctness.

Reply:
Reference for example?

Reply:
From where did you use these examples as test? We need references for verifying correctness.

Reply:
We have built our test cases, there is no exact reference for it. For your convenience, we can make a document that will contain a pictorial representation of the test cases handled.

Reply:
Visualisations for RedBlackTrees are possible through already existing sources(see README for some examples). You can verify the correctness of your code by comparing your outputs with them.